internal/archive/archive_test.go

```diff
@@ -492,44 +519,122 @@ func read(r io.Reader) string {
 
 // ----------------------------------------------------------------------------------------
 // Real archive tests, only enabled via --real-archive.
+//
+// Additionally, run tests against Ubuntu Pro archives by passing --pro-archive.
+// The host machine must be Pro enabled and relevant Pro services must be
+// enabled. The following commands might help:
+// 		sudo pro attach <pro-token> --no-auto-enable
+//		sudo pro enable fips fips-updates esm-apps esm-infra --assume-yes
 
 var realArchiveFlag = flag.Bool("real-archive", false, "Perform tests against real archive")
+var proArchiveFlag = flag.Bool("pro-archive", false, "Perform tests against real Ubuntu Pro archive")
 
 func (s *S) TestRealArchive(c *C) {
 	if !*realArchiveFlag {
 		c.Skip("--real-archive not provided")
 	}
-	for _, release := range ubuntuReleases {
-		for _, arch := range elfToDebArch {
-			s.testOpenArchiveArch(c, release, arch)
+	allArch := make([]string, 0, len(elfToDebArch))
+	for _, arch := range elfToDebArch {
+		allArch = append(allArch, arch)
+	}
+	for _, test := range realArchiveTests {
+		if test.pro != "" && !*proArchiveFlag {
+			// --pro-archive is not provided. Ignore test.
+			continue
+		}
+		if len(test.architectures) == 0 {
```

Comment:
Think about splitting the real-archive non-pro tests and the pro tests for better visibility.